src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -198,23 +201,76 @@ def load_arguments(self, _):
         c.argument('operation_ids', nargs='+', help='A list of operation ids to show')
 
     with self.argument_context('deployment') as c:
-        c.argument('deployment_name', options_list=['--name', '-n'], required=True, help='The deployment name.')
+        c.argument('deployment_name', arg_type=deployment_name_type)
         c.argument('deployment_location', arg_type=get_location_type(self.cli_ctx), required=True)
-        c.argument('template_file', completer=FilesCompleter(), type=file_type, help="The template file path in the file system")
-        c.argument('template_uri', help='The URI to a remote template file')
-        c.argument('parameters', action='append', nargs='+', completer=FilesCompleter())
+        c.argument('template_file', arg_type=deployment_template_file_type)
+        c.argument('template_uri', arg_type=deployment_template_uri_type)
+        c.argument('parameters', arg_type=deployment_parameters_type)
 
     with self.argument_context('deployment create') as c:
-        c.argument('deployment_name', options_list=['--name', '-n'], required=False,
-                   help='The deployment name. Default to template file base name')
+        c.argument('deployment_name', arg_type=deployment_create_name_type)
         c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
 
     with self.argument_context('deployment validate') as c:
+        c.argument('deployment_name', arg_type=deployment_create_name_type)
         c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
 
-    with self.argument_context('deployment operation show') as c:
+    with self.argument_context('deployment operation') as c:
         c.argument('operation_ids', nargs='+', help='A list of operation ids to show')
```

Comment:
@Tiano2017 I found a new question: Is this `operation_ids` is a parameter of the entire scope for `deployment operation` now? But the description on help looks like it's just a parameter only for `deployment operation show`.

Reply:
The `operation_ids` is only used for the `show` commands. It's put here so that it's accessible to all the `show` commands: we have `deployment operation show`, `deployment operation sub show`, `deployment operation group show`, `deployment operation mg show`, `deployment operation tenant show`.

Reply:
OK, it's fine